plaso/engine/artifacts_filter_file.py

```diff
@@ -0,0 +1,272 @@
+from dfwinreg import registry_searcher
+
+
+from plaso.lib import py2to3
+from plaso.lib import errors
+
+
+ARTIFACTS_FILTER_FILE = 'ARTIFACTS_FILTER_FILE'
+COMPATIBLE_DFWINREG_KEYS = [r'HKEY_LOCAL_MACHINE',
+                            r'HKEY_LOCAL_MACHINE\SYSTEM',
+                            r'HKEY_LOCAL_MACHINE\SOFTWARE',
+                            r'HKEY_LOCAL_MACHINE\SAM',
+                            r'HKEY_LOCAL_MACHINE\SECURITY']
+RECURSIVE_GLOB_LIMIT = 10
+
+
+class ArtifactsFilterFile(object):
+  """Forensic artifacts file.
+
+  A forensic artifacts file contains one or more forensic artifacts filters.
+
+  Forensic artifacts are defined in:
+  https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc
+  """
+
+  def __init__(self, path, knowledge_base):
+    """Initializes a filter file.
+
+    Args:
+      path (str): path to a file that contains one or more forensic artifacts.
+      knowledge_base (KnowledgeBase): Knowledge base for Log2Timeline.
+    """
+    super(ArtifactsFilterFile, self).__init__()
+    self._path = path
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Build find specification from a forensic artifacts file.
+
+    Args:
+      environment_variables (Optional[list[EnvironmentVariableArtifact]]):
+          environment variables.
+    """
+    find_specs = {}
+    path_attributes = self._BuildPathAttributes(environment_variables)
+    artifact_registry = artifacts_registry.ArtifactDefinitionsRegistry()
+    artifact_reader = artifacts_reader.YamlArtifactsReader()
+
+    try:
+      artifact_registry.ReadFromFile(artifact_reader, self._path)
+
+    except (KeyError, artifacts_errors.FormatError) as exception:
+      raise errors.BadConfigOption((
+          'Unable to read artifact definitions from: {0:s} with '
+          'error: ''{1!s}').format(self._path, exception))
+
+    for definition in artifact_registry.GetDefinitions():
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(path_entry, source.separator,
+                                                path_attributes, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(key_entry,
+                                                      path_attributes,
+                                                      find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
```

Comment:
not sure what you mean here

Reply:
A TODO to follow-up and see if there are plans for dfwinreg to support FindSpecs that specify a key-value pair, instead of the just the key path.  Right now I convert all key-value pairs to key paths only to build the FindSpecs.